pkg/bundle/manager.go

```diff
@@ -109,6 +109,28 @@ func (m *bundleManager) ProcessBundleController(ctx context.Context, pbc *api.Pa
 			return fmt.Errorf("creating namespace for %s: %s", pbc.Name, err)
 		}
 
+		activeBundle, err := m.bundleClient.GetBundle(ctx, pbc.Spec.ActiveBundle)
+		if err != nil {
+			m.log.Error(err, "Unable to get active bundle", "bundle", pbc.Spec.ActiveBundle)
+			return nil
+		}
+
+		if activeBundle == nil {
```

Comment:
You don't handle the case where activeBundle isn't nil, but also isn't set, as would be returned from GetBundle(ctx, "") in pkg/bundle/client.go:148. You return an initialized value there, so it won't match as nil.

Also, I don't see a test below for that scenario, though there might be an existing test that I can't see cuz github.

Reply:
Changed this to something more logical